cmd/prompt.go

```diff
@@ -0,0 +1,402 @@
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	goprompt "github.com/c-bata/go-prompt"
+	"github.com/c-bata/go-prompt/completer"
+	homedir "github.com/mitchellh/go-homedir"
+	"github.com/nsf/termbox-go"
+	"github.com/openconfig/goyang/pkg/yang"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+)
+
+var promptMode bool
+var promptHistory []string
+var schemaTree *yang.Entry
+var promptModeCmd = &cobra.Command{
+	Use:     "prompt-mode",
+	Aliases: []string{"prompt"},
+	Short:   "enter the interactive gnmic prompt-mode",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := pathCmdRun(promptDirs, promptFiles, promptExcluded, true)
+		if err == nil {
+			promptMode = true
+		}
+		// load history
+		promptHistory = make([]string, 0, 256)
```

Comment:
shouldn't this capacity match the history size written (128)?

Reply:
The 128 commands in history is just saved and loaded. But, I'd created larger string slice for new typed commands. I'd heard that the append() function of pre-allocated go slice (`make([]string, 0, 256`) has higher performance than empty slice (`make([]string)`).

Reply:
Yes agree with allocating capacity first.
I think this can be improved later on by limiting the size to 128 when appending.